### PR TITLE
Update Scenario Selector documentation with 4.0 features

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Addresses #1213 and a bit more.

This should partially be backported to 4.0.

- Added Filtering, Sorting and Searching section in the Scenario Selector doc page.
- Added actual MUI icons to be used in the doc body text (added to the table control doc page).
- Tiny doc adjustement in the metric control formatting paragraph.
- Tiny doc change in the toggle control.
- Force UTF-8 encoding in Markdown generation.
- Release Notes updated.

### CODE REVIEW

File: docs/refmans/gui/viselements/corelements/scenario_selector.md_template [hunk 1]
```diff
@@ -1,55 +1,80 @@
...
 # Details
 
-The scenario selector displays a tree selector where scenarios are grouped based on their cycle
-(if the property [*display_cycles*](#p-display_cycles) has not been set to False).<br/>
-If the [*show_primary_flag*](#p-show_primary_flag) property has not been forced to False, the
-label of the primary scenario are overlaid with a small visual hint that lets users spot them
-immediately.
+The Scenario Selector is a visual control that displays a tree structure where scenarios are grouped
+by their respective cycles (provided the [*display_cycles*](#p-display_cycles) property is not set)
+to False.
```
Comment (FredLL-Avaiga) [inline docs/refmans/gui/viselements/corelements/scenario_selector.md_template]: move parenthesis ?

File: docs/refmans/gui/viselements/corelements/scenario_selector.md_template [hunk 2]
```diff
@@ -58,16 +83,21 @@ Here is how this dialog box looks like:
...
-A custom property can be removed by pressing the trash button next to it.<br/>
+In the dialog, users can:
 
-The 'Cancel' button closes the dialog without changing anything.<br/>
-The 'Delete' button deletes the edited scenario.
+<ul>
+<li>Modify the Scenario Label,</li>
+<li>Add custom properties by filling in the 'Key' and 'Value' fields in the 'Custom Properties'
+    section then pressing the [MUI:Add] icon (Add),</li>
+<li>Remove custom properties by click the [MUI:DeleteOutline] icon (Delete) next to an existing
```
Comment (FredLL-Avaiga) [inline docs/refmans/gui/viselements/corelements/scenario_selector.md_template]: clicking ?

File: docs/refmans/gui/viselements/corelements/scenario_selector.md_template [hunk 3]
```diff
@@ -77,6 +107,157 @@ application can then use the selected value.
...
+
+By default, the Scenario Selector sorts scenarios in ascending order by their creation date. When
+applicable, scenarios are grouped by their respective Cycle.
+
+Users can customize the sorting order by defining custom sorting criteria, similar to the process of
+filtering. This allows users to create a prioritized sequence of sorting rules.
+
+Each sorting criterion specifies:
+
+- The scenario  property to sort by (e.g., label, creation date).
```
Comment (FredLL-Avaiga) [inline docs/refmans/gui/viselements/corelements/scenario_selector.md_template]: double space between scenario and property (not that it matters in any way)

File: docs/refmans/gui/viselements/generic/metric.md_template [hunk 4]
```diff
@@ -137,7 +137,12 @@ Here is the control definition:
     delta_format=%d %%
 
 The syntax for the [*format*](#p-format) and [*delta_format*](#p-delta_format) properties is the one
-used by the native `sprintf()` function. In this case, values are both integers.
+used by the native `sprintf()` function. In this case, values are both integers.<br/>
+Note that in this example, we want to *delta* value to appear with a percent sign suffix. To get
```
Comment (FredLL-Avaiga) [inline docs/refmans/gui/viselements/generic/metric.md_template]: `the *delta*` ?

File: docs/userman/gui/extension/dynamic_element/tabular_data_props.md [hunk 5]
```diff
@@ -1,32 +1,37 @@
...
+their properties to
+specify the type of data they support, similar to scalar properties, but now using a structure
+suitable for collections.
+This is handled by the PropertyType class, where you can define the type as an array or table
+format, enabling the binding of multidimensional data.
+
+For example, a table element can bind a two-dimensional array or a list of objects to its
+properties. Each time the data in the array changes, the table element automatically refreshes to
+display the updated content in the user interface.<br/>
+This approach leverages TypeScript and JavaScript code in the backend, working with React to
```
Comment (FredLL-Avaiga) [inline docs/userman/gui/extension/dynamic_element/tabular_data_props.md]: I don't understand this sentence.

**Review by FredLL-Avaiga — APPROVED** [review]

File: docs/refmans/gui/viselements/corelements/scenario_selector.md_template (hunk 1 above)
Comment (FabienLelaquais) [reply docs/refmans/gui/viselements/corelements/scenario_selector.md_template]: 👍

File: docs/refmans/gui/viselements/corelements/scenario_selector.md_template (hunk 2 above)
Comment (FabienLelaquais) [reply docs/refmans/gui/viselements/corelements/scenario_selector.md_template]: 👍

File: docs/refmans/gui/viselements/generic/metric.md_template (hunk 4 above)
Comment (FabienLelaquais) [reply docs/refmans/gui/viselements/generic/metric.md_template]: 👍

File: docs/refmans/gui/viselements/corelements/scenario_selector.md_template (hunk 3 above)
Comment (FabienLelaquais) [reply docs/refmans/gui/viselements/corelements/scenario_selector.md_template]: 👍

File: docs/userman/gui/extension/dynamic_element/tabular_data_props.md (hunk 5 above)
Comment (FabienLelaquais) [reply docs/userman/gui/extension/dynamic_element/tabular_data_props.md]: Rewritten. Thx

**Review by namnguyen20999 — APPROVED** [review]

**Review by FredLL-Avaiga — APPROVED** [review]
🥇

**Pull request merged**